### ISSUE TITLE
Compiler warnings on CCL reveal bugs...

### ISSUE THREAD

**Issue by nklein** [eugeneia/maxpc#4]
There are two compiler warnings on CCL.  The first is `input.lisp`

``` common-lisp
(defmethod input-sequence (input (length integer))
  (loop for i from 1 to length
        for input = input then (input-rest input)
     collect (input-first input)))
```

On both SBCL and CCL, this results in code where the `input` used as the initial value of `input` in the loop is not the outer value of the variable `input` but the value `NIL`.

The other is in `test.lisp` in the `test-digit` function, one of the tests does not actually compare against the `result`.

``` common-lisp
  (passert "-f423" (=integer-number 16) (= #x-f423))
```

**Comment by nklein** [comment]
Apparently, there's been much debate about the first issue and it was left to the implementation:
http://www.lispworks.com/documentation/lw51/CLHS/Issues/iss222_w.htm

**Comment by eugeneia** [comment]
Thanks for reporting this, great catch! Fixed in #5.
